src/wpool/mongoose_gun_worker.erl

```diff
@@ -0,0 +1,262 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% This module a wpool worker capable of establishing HTTP/2
+%%% connections using the `Gun' library.
+%%% @end
+%%%-------------------------------------------------------------------
+-module(mongoose_gun_worker).
+
+-behaviour(gen_server).
+
+%% API
+-export([start_link/2]).
```

Comment:
I actually think that this function is never used, wpool seems to be doing himself the `     gen_server:start_link(ModuleName, Args, []).` with what was given to him in `mongoose_wpool_http:wpool_spec/2`, the tuple `{mongoose_gun_worker, {TargetServer, HttpOpts}}`, the first element is precisely `ModuleName`, and the second is `Args`.